### PR TITLE
Adding encoding to import function

### PR DESCRIPTION
**Pull request by netmonk**
After defining a latin1 database, there are error when importing sql scripts encoded in latin1 charset with the import function. 

`mysql` has an option `--default-character-set` to specify the encoding of the input file. This commit modify the import function so that if `encoding` exists when using the import function, it will add `--default-character-set="value of encoding"` to the options of mysql command.  This way encoded file in latin1 for example will be imported without error (by default, the file is considered utf8 encoded)

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
mysql_db supporting import respecting provided `encoding` value. 

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by netmonk** [comment]
Anyone reviewing ? @Alexander198961  @Andersson007  @Xyon  @bmalynovytch  @bmildren  
 @kurtdavis  @michaelcoburn @Oneiroi @tolland

**Comment by bmalynovytch** [comment]
Hi @netmonk !
Thank you for your contribution.
I'm gonna give a look at it.
Nobody reviewed it as you forgot to add the label saying you were done with your changes.
You'll find the infos here : https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#for-pull-request-submitters

**Review by bmalynovytch — CHANGES_REQUESTED** [review]
You need to add the corresponding documentation, as well as an example.
Good job though 👍

**Comment by netmonk** [comment]
ansibot where are you ?

File: lib/ansible/modules/database/mysql/mysql_db.py [hunk 1]
```diff
@@ -117,12 +117,21 @@
...
     state: import
     name: all
     target: /tmp/dump.sql
 
+# Import of sql script with encoding option
+- name: Import dump.sql with specific latin1 encoding, similar to mysql -u <username> --default-character-set=latin1 -p <password> < dump.sql
+  mysql_db:
+    state: import
+    name: all
+    encoding: "latin1"
```
Comment (bmalynovytch) [inline lib/ansible/modules/database/mysql/mysql_db.py]:
```suggestion
    encoding: latin1
```

**Review by bmalynovytch — CHANGES_REQUESTED** [review]

**Comment by bmalynovytch** [comment]
@netmonk when you're done with your changes, just type `ready for review` (replace spaces by underscores) in the chat the let the bot know you're done.

**Comment by netmonk** [comment]
ready_for_review

File: lib/ansible/modules/database/mysql/mysql_db.py [hunk 2]
```diff
@@ -117,12 +117,21 @@
     name: all
     target: /tmp/dump.sql
 
-- name: Import file.sql similar to mysql -u <username> -p <password> < hostname.sql
+# Basic import of sql script file
```
Comment (bmalynovytch) [inline lib/ansible/modules/database/mysql/mysql_db.py]: For conciseness of the examples, maybe should we keep only 1 example with multiple option enabled / shown.
I'd keep only your example

**Review by bmalynovytch — CHANGES_REQUESTED** [review]

**Comment by netmonk** [comment]
ready_for_review

**Comment by bmalynovytch** [comment]
@netmonk you marked my review as fixed but it's not and you don't even express your point of view
I won't review further

**Comment by netmonk** [comment]
> @netmonk you marked my review as fixed but it's not and you don't even express your point of view
> I won't review further

well, i prefer to have multiple exemples, like there are multiples exemples for others functions of the module. 
Often the Ansible documention is really unclear about usable options when using a module. I prefer to have two exemples with two use-cases specific use case. 
Is it clearer ?

**Comment by bmalynovytch** [comment]
The `import` function remains the same.
Specifying the encoding isn't a different function, just an argument.

**Comment by netmonk** [comment]
This argument is optionnal, and to be used it needed.

**Review by bmalynovytch — APPROVED** [review]

**Comment by bmalynovytch** [comment]
LGTM
shipit

**Comment by netmonk** [comment]
Do i need to close the pull request now ?

**Comment by bmalynovytch** [comment]
No it’ll get closed when merged. Before that other contributors must review and add `shipit` command to confirm they are ok with your code.

**Comment by bmalynovytch** [comment]
bot_status

**Review by Andersson007 — COMMENTED** [review]
@netmonk , thank you for the PR!
Would be cool to see ci test for this case added to the existing tests here ```test/integration/targets/mysql_db/tasks/```.
Could you please make them.
Then we can merge it without any concerns (IMO ci should be added to all prs except the most primitive and doc related).

**Comment by netmonk** [comment]
I will try to do it, need some more time for it. When thinking about it, i just realised that dump is also prone to charset specification. So i can also update the dump function to take `encoding` into consideration. What do you think about ?

**Comment by Andersson007** [comment]
I don’t know :) i can’t think anything because i’m not a hardcore MySQL dba, we use it for a couple of non production services for legacy reasons. So I can look at this only in general. Anyway I believe we need tests for every bugfux/new feature PR not to break anything existent by merging them. Would be cool to be sure that all related options and possible cases that might be affected by the code changes are covered.

**Comment by netmonk** [comment]
Im not a dba neither. It's just i just spend the last 6 months advocating against the use of the shell module in Ansible in other dpt of my client company. And then i was asking to automate deployement of a third party software where the mysql scripts are encoded in Latin1 and i had some issue. That is why i decided to look at code and fix it. In order to avoid to use the shell module... :)

**Comment by Andersson007** [comment]
ok, @bmalynovytch approved it, so, if there were ci tests I would approve it too :)

**Comment by netmonk** [comment]
will do. gimme time ! :)

**Comment by netmonk** [comment]
Can someone destroy and delete this PR... The rebase is not welcome, time to start again from fresh fork... thank you !

**Pull request closed without merging**